systemtest/src/main/java/io/strimzi/systemtest/resources/crd/kafkaclients/KafkaAdminClient.java

```diff
@@ -0,0 +1,259 @@
+/*
+ * Copyright Strimzi authors.
+ * License: Apache License 2.0 (see the file LICENSE or http://apache.org/licenses/LICENSE-2.0.html).
+ */
+package io.strimzi.systemtest.resources.crd.kafkaclients;
+
+import io.fabric8.kubernetes.api.model.batch.v1.JobBuilder;
+import io.strimzi.systemtest.Constants;
+import io.strimzi.systemtest.Environment;
+import io.strimzi.systemtest.resources.ResourceManager;
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+
+import java.security.InvalidParameterException;
+import java.util.HashMap;
+import java.util.Map;
+
+/**
+ * KafkaAdmin Client - initial implementation of test-client,
+ * which supports Topic management operations only.
+ * Create, delete, update topics and partitions increase (in bulk, with offsets).
+ */
+public class KafkaAdminClient {
+
+    private static final Logger LOGGER = LogManager.getLogger(KafkaAdminClient.class);
+
+    protected String adminName;
+    protected String bootstrapAddress;
+    protected String topicName;
+    protected int topicCount;
+    protected int topicOffset;
+    protected int partitions;
+    protected int replicationFactor;
+    protected String topicOperation;
+    protected String additionalConfig;
+    protected String namespaceName;
+
+    protected KafkaAdminClient(KafkaAdminClient.Builder builder) {
+        if (builder.topicOperation == null || builder.topicOperation.isEmpty())
+            throw new InvalidParameterException("TopicOperation must be set.");
+        if (builder.bootstrapAddress == null || builder.bootstrapAddress.isEmpty())
+            throw new InvalidParameterException("Bootstrap server is not set.");
+        if ((builder.topicName == null || builder.topicName.isEmpty()) && !(builder.topicOperation.equals("help") || builder.topicOperation.equals("list")))
+            throw new InvalidParameterException("Topic name (or 'prefix' if topic count > 1) is not set.");
+
+        replicationFactor = (builder.replicationFactor == 0) ? 1 : builder.replicationFactor;
+        partitions = (builder.partitions == 0) ? 1 : builder.partitions;
```

Comment:
What If I set a negative number of partitions? Same scenario with replication factor or topic count..

Reply:
It will go *boom* :-)  
I will change it to `<= 0`, so default 1 is used in this case.